src/sentry/web/frontend/idp_email_verification.py

```diff
@@ -32,3 +26,27 @@ def handle(self, request: Request, key: str) -> HttpResponse:
                 "sentry/idp_account_verified.html", context=context, request=request
             )
         return render_to_response("sentry/idp_account_not_verified.html", request=request)
+
+    @staticmethod
+    def _recover_org_member(verification_value):
+        member_id = verification_value.get("member_id")
+        if member_id is None:
+            return None  # the user was not an org member when the record was written
+        try:
+            return OrganizationMember.objects.get(id=member_id)
+        except OrganizationMember.DoesNotExist:
+            return None  # the user has left the org since the record was written
+
+    @staticmethod
```

Comment:
nit: is it cleaner to make this a `@classmethod` so you can use `cls` below instead of having to reference `AccountConfirmationView`?

Reply:
Good idea, but obviated if we skip the org member query.